osu.Game/Screens/Ranking/Statistics/HitEventTimingDistributionGraph.cs

```diff
@@ -64,10 +64,22 @@ private void load()
             // Prevent div-by-0 by enforcing a minimum bin size
             binSize = Math.Max(1, binSize);
 
+            bool roundUp = true;
+
             foreach (var e in hitEvents)
             {
-                int binOffset = (int)Math.Round(e.TimeOffset / binSize, MidpointRounding.AwayFromZero);
-                bins[timing_distribution_centre_bin_index + binOffset]++;
+                double binOffset = e.TimeOffset / binSize;
+
+                // .NET's round midpoint handling doesn't provide a behaviour that works amazingly for display
+                // purposes here. We want midpoint rounding to roughly distribute evenly to each adjacent bucket
+                // so the easiest way is to cycle between downwards and upwards rounding as we process events.
+                if (Math.Abs(binOffset - (int)binOffset) == 0.5)
+                {
+                    binOffset += Math.Sign(binOffset) * (roundUp ? 1 : 0);
+                    roundUp = !roundUp;
+                }
+
+                bins[timing_distribution_centre_bin_index + (int)binOffset]++;
```

Comment:
I think this change is doing more than what it intended to. Compare `TestAroundCentre` in the test scene:

| `master` | this PR |
| :-: | :-: |
| ![2022-03-03-191745_752x246_scrot](https://user-images.githubusercontent.com/20418176/156627191-5100abae-113e-40df-b24c-99a36003bb82.png) | ![2022-03-03-191800_684x228_scrot](https://user-images.githubusercontent.com/20418176/156627200-cf9b38e5-6b8e-43ec-a1b7-58e0c00234d7.png) |

The test is using 300 points, uniformly distributed around 0. So there definitely shouldn't be a spike, but with this PR there is one, because the `Math.Round()` call was removed entirely - if `binOffset` is 0.7, for instance, it'll be truncated down to 0 here rather than rounded up to 1 as previously. I presume the intention was to only handle the midpoint case.

To get my point across better, I'll leave a table here that shows how the rounding behaves:

| fractional part | [0.0, 0.5) | 0.5 | (0.5, 1] |
| :- | :-: | :-: | :-: |
| `master` | 0 | 1 | 1 |
| this PR | 0 | 0 or 1, evenly distributed | 0 |
| what I assume was desired | 0 | 0 or 1, evenly distributed |

If my assumptions are correct, below is an example patch that achieves that result:

```diff
diff --git a/osu.Game/Screens/Ranking/Statistics/HitEventTimingDistributionGraph.cs b/osu.Game/Screens/Ranking/Statistics/HitEventTimingDistributionGraph.cs
index 3a4641cba9..235eac7f78 100644
--- a/osu.Game/Screens/Ranking/Statistics/HitEventTimingDistributionGraph.cs
+++ b/osu.Game/Screens/Ranking/Statistics/HitEventTimingDistributionGraph.cs
@@ -75,11 +75,11 @@ private void load()
                 // so the easiest way is to cycle between downwards and upwards rounding as we process events.
                 if (Math.Abs(binOffset - (int)binOffset) == 0.5)
                 {
-                    binOffset += Math.Sign(binOffset) * (roundUp ? 1 : 0);
+                    binOffset = (int)binOffset + Math.Sign(binOffset) * (roundUp ? 1 : 0);
                     roundUp = !roundUp;
                 }
 
-                bins[timing_distribution_centre_bin_index + (int)binOffset]++;
+                bins[timing_distribution_centre_bin_index + (int)Math.Round(binOffset, MidpointRounding.AwayFromZero)]++;
             }
 
             int maxCount = bins.Max();
```

Reply:
Nice catch, I knew something was off..